Miscellaneous updates

vicencb opened the pull request:
Hello,
i am using mussel with all those changes.
Please, feel free to pick the ones you like.

Regards,
  Vicente.

firasuke commented:
Thanks for using `mussel` and for sharing your patches. I'll review them shortly..

firasuke reviewed (commented):
`mussel` no longer depends on `rsync` and the target `headers_install` was replaced with commands with identical functionality.

firasuke commented on mussel:
mussel no longer depends on rsync and the target headers_install was replaced with commands with identical functionality.

firasuke replied:
The optional C++ support patch could be accepted.

firasuke replied:
Consider adding `mips32` and `mips32r2` as aliases to `mips`. Also check if the following options are still valid:

```sh
--with-mips-plt
--with-arch-32=mips2
--with-tune-32=mips32
--with-fp-32=32
```

Also try to prefer hard float over soft float if it's supported.

firasuke replied:
Well making checksums optional would work if you're modifying the script for updates and such, which is expected from mussel users.

firasuke replied:
I agree with the `lib32` removal. It was added by some older patch, but I don't think anyone is using/relying on this directory anymore..

takusuman commented on mussel:
```diff
@@ -75,7 +75,7 @@ if [ "$checksum_command" = "b3sum" ]; then
   gmp_sum=fffe4996713928ae19331c8ef39129e46d3bf5b7182820656fd4639435cd83a4
   isl_sum=a419cf24561a34f81ec46d251b25d5aaccedbbf87ed2b96f3f0fb17a79a8c913
-  linux_sum=4082a33166c9a670036576d096bb37d499f6dd2d69d1c9da54aa4105e47791c5
+  linux_sum=
```
Wait, you plainly removed the checksums?
That may cause some fuss later...

takusuman commented on mussel:
```diff
@@ -727,6 +746,9 @@ printf "Optional pkg-config Support:    $PKG_CONFIG_SUPPORT\n"
 printf "Optional Quadmath Support:      $QUADMATH_SUPPORT\n"
 printf "Optional custom vendor:         '$CUSTOM_VENDOR'\n\n"
 
+# fail early if rsync is needed but not found
```
As @firasuke said, if I'm not erred, rsync is not a dependency of kbuild anymore.

takusuman commented on mussel:
```diff
@@ -743,23 +765,26 @@ printf "PATH: \"$PATH\"\nMAKE: \"$MAKE\"\n\n" >>"$MLOG"
-mpackage gcc "$gcc_url" $gcc_sum $gcc_ver
-mpackage gmp "$gmp_url" $gmp_sum $gmp_ver
-[ $ISL_SUPPORT = "yes" ] && mpackage isl "$isl_url" $isl_sum $isl_ver
+mpackage binutils "$binutils_url" "$binutils_sum" $binutils_ver
```
I don't want to be _that_ type of shell guy but I must say that there's no need to quote checksums since we do not expect it to have spaces or anything --- I don't think it would be even a micro-optimzation, just a pure formality. I know you've unset the ``$linux_sum`` variable and made it empty, but it also shouldn't be this way.

vicencb commented:
@firasuke:
> mussel no longer depends on rsync and the target headers_install was replaced with commands with identical functionality.

Removed rsync check.

> The optional C++ support patch could be accepted.

I had an extra comma and it worked for me, but for correctness, i removed it.

> Consider adding `mips32` and `mips32r2` as aliases to `mips`.

Done.

> Also check if the following options are still valid:
> 
> ```shell
> --with-mips-plt
> --with-arch-32=mips2
> --with-tune-32=mips32
> --with-fp-32=32
> ```

From the documentation [here](https://gcc.gnu.org/onlinedocs/gcc/MIPS-Options.html) i only see the `plt`-related one, which only changes the default value among `-mplt` and `-mno-plt`. If you don't mind, i prefer to omit it and let it be by default.

The other options are not documented. What are they good for?

> Also try to prefer hard float over soft float if it's supported.

I am targetting the AR9341 and MT7688 CPUs and neither have FPU, so, hard-float is not supported.

> Optimizing for size could be added as a separate option if needed.

It is now an option.

@takusuman:
> Wait, you plainly removed the checksums?

Yes and no: this patch is not intended to be applied, it only shows that once checksums are optional it would be that easy for an end user to update versions.

firasuke understood my intention given his comment:
> Well making checksums optional would work if you're modifying the script for updates and such, which is expected from mussel users.

> rsync is not a dependency of kbuild anymore

Already addressed from firasuke's comment.

> I must say that there's no need to quote checksums since we do not expect it to have spaces or anything

The quotation is needed because it could be the empty string.

vicencb commented:
If this pull request if finally accepted, please, don't add me as a contributor into the front page.

firasuke commented:
Looks good to me. Merged!

> If this pull request if finally accepted, please, don't add me as a contributor into the front page.

As you wish, but I'd prefer if you were added.

takusuman commented:
O.k., great. @vicencb
I forgot to approve your changes (even though it wasn't really necessary since the right of veto comes from @firasuke).

takusuman commented:
My honest opinion @firasuke: I think it would be great to have at the log file if the file has no checksum. By the way, we would need to update the README to include the last two parameters implemented.

firasuke commented:
> O.k., great. @vicencb I forgot to approve your changes (even though it wasn't really necessary since the right of veto comes from @firasuke).

Oh feel free to critique any change, as you've contributed a lot to `mussel`.

> My frank opinion @firasuke: I think it would be great to have at the log file if the file has no checksum. By the way, we would need to update the README to include the last two parameters implemented.

Sure, to be honest the script needs some refactoring and a bit of cleaning. Also some functions could be made simpler by assuming the existence of commonly found tools, but we'll leave that for another day and another issue :)

takusuman commented:
> Sure, to be honest the script needs some refactoring and a bit of cleaning.

I thought we could try to make the option list printing a little less redundant. Also, we could try to use ``printf``'s 'f' instead of mixing variables amid the string --- or maybe just use echo? I know that velocity on running a script isn't a concern when having to compile GCC, but I honestly think it makes more sense than printf currently.
